test/integration/services/modal.js

```diff
@@ -0,0 +1,182 @@
+// import _ from 'underscore';
+
+context('Modal Service', function() {
+  const smallOnTall = () => {
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:tall', {
+          headingText: 'Tall Modal Header',
+          bodyText: 'Tall Modal, which is under the small modal.',
+        });
+      });
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Small Modal Header',
+          bodyText: 'Small Modal, this guy shows over the top of the tall modal.',
+        });
+      });
+  };
+  specify('Displaying', function() {
+    const buttonStub = cy.stub();
+    cy
+      .visit('/');
+
+    smallOnTall();
+
+    cy
+      .get('.modal--small')
+      .should('contain', 'Small Modal')
+      .find('.modal-cancel')
+      .click();
+
+    cy
+      .get('.modal--tall')
+      .should('contain', 'Tall Modal')
+      .find('.modal-cancel')
+      .click();
+
+    // Close using the X icon
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Small Modal Header',
+          bodyText: 'Small Modal',
+          onCancel: buttonStub,
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .find('.modal-close')
+      .click()
+      .then(() => {
+        expect(buttonStub).to.be.calledOnce;
+        buttonStub.reset();
+      });
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Close On Save',
+          bodyText: 'Default onSave behavior. Also a cool custom class!',
+          buttonClass: 'save-button',
+          className: 'modal--small custom-class',
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .should('have.class', 'custom-class')
+      .find('.save-button')
+      .click();
```

Comment:
changed this one to handle the default, and let the stub take care of the custom